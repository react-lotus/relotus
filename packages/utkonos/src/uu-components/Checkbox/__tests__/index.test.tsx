/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import type { Ref } from 'react';
import { createRef } from 'react';
import { Checkbox, CheckboxProps } from '../index';

const CheckboxWrapper = (props: Partial<CheckboxProps> & { customRef?: Ref<HTMLInputElement> }) => {
  const { customRef, ...rest } = props;
  return <Checkbox ref={customRef} {...rest} />;
};

describe('Uikit/Checkbox', () => {
  describe('snapshot', () => {
    it('default', () => {
      const component = <Checkbox />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('checked', () => {
      const component = <Checkbox defaultChecked />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('disabled', () => {
      const component = <Checkbox disabled />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('disabled checked', () => {
      const component = <Checkbox disabled defaultChecked />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('component', () => {
    it('should properly pass ref to the Checkbox element', () => {
      const ref = createRef<HTMLInputElement>();
      render(<CheckboxWrapper customRef={ref} />);
      ref.current?.focus();
      const inputCheckbox = screen.getAllByRole('checkbox');
      expect(inputCheckbox[0]).toHaveFocus();
    });
  });
});
