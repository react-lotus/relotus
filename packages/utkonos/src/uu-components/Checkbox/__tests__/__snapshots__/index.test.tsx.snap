// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uikit/Checkbox snapshot checked 1`] = `
<label
  className="uu-checkbox"
  htmlFor="nanoid"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    className="uu-checkboxInput"
    defaultChecked={true}
    id="nanoid"
    type="checkbox"
  />
  <span
    aria-checked={false}
    className="uu-checkboxMark"
    role="checkbox"
  />
</label>
`;

exports[`Uikit/Checkbox snapshot default 1`] = `
<label
  className="uu-checkbox"
  htmlFor="nanoid"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    className="uu-checkboxInput"
    id="nanoid"
    type="checkbox"
  />
  <span
    aria-checked={false}
    className="uu-checkboxMark"
    role="checkbox"
  />
</label>
`;

exports[`Uikit/Checkbox snapshot disabled 1`] = `
<label
  className="uu-checkbox uu-checkbox_disabled"
  htmlFor="nanoid"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    className="uu-checkboxInput"
    disabled={true}
    id="nanoid"
    type="checkbox"
  />
  <span
    aria-checked={false}
    className="uu-checkboxMark uu-checkboxMark_disabled"
    role="checkbox"
  />
</label>
`;

exports[`Uikit/Checkbox snapshot disabled checked 1`] = `
<label
  className="uu-checkbox uu-checkbox_disabled"
  htmlFor="nanoid"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    className="uu-checkboxInput"
    defaultChecked={true}
    disabled={true}
    id="nanoid"
    type="checkbox"
  />
  <span
    aria-checked={false}
    className="uu-checkboxMark uu-checkboxMark_disabled"
    role="checkbox"
  />
</label>
`;
